### PR TITLE
[Digital Ocean] Implement Delete Instance logic for rolling update

### PR DESCRIPTION
**Pull request by srikiz**
Implement delete instance for handling rolling updates.

### CODE REVIEW

**Comment by olemarkus** [comment]
Have a look at https://github.com/kubernetes/kops/pull/9927
You may want to merge this into your PR so that rolling update does not require running anything in a second terminal. I think this should work fairly out of the box also with DO.

What I do there is basically just run the full update process in-between terminations.

**Comment by srikiz** [comment]
@olemarkus  - Would it be okay if I merge this separately as a new PR once #9927 is merged?

**Comment by olemarkus** [comment]
Fine for me :)

**Comment by hakman** [comment]
In this case, let's try to get #9927 moving. @zetaab do you have some time to look at it?
/ok-to-test
/hold until #9927 is merged

**Comment by hakman** [comment]
#9927 has been merged.
/hold cancel

File: pkg/instancegroups/instancegroups.go [hunk 1]
```diff
@@ -387,7 +387,8 @@ func (c *RollingUpdateCluster) drainTerminateAndWait(u *cloudinstances.CloudInst
 }
 
 func (c *RollingUpdateCluster) reconcileInstanceGroup() error {
-	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack {
+	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack ||
+		api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderDO {
```
Comment (hakman) [inline pkg/instancegroups/instancegroups.go:391]: Always true, maybe you wanted AND?
```suggestion
	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack &&
		api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderDO {
```

Comment (srikiz) [reply pkg/instancegroups/instancegroups.go:391]: Good catch, thanks @hakman

**Comment by olemarkus** [comment]
In `instancegroups.go` there is a function `reconcileInstanceGroup()`. If you add a check for DO there, rolling updates of DO clusters should work too.

**Comment by srikiz** [comment]
> In `instancegroups.go` there is a function `reconcileInstanceGroup()`. If you add a check for DO there, rolling updates of DO clusters should work too.

Okay, I have actually added the check in reconcileInstanceGroup, am I missing something?

**Comment by olemarkus** [comment]
Sorry, I must have missed that.
/lgtm

**Pull request merged**